tango/executor.py

```diff
@@ -24,30 +27,26 @@ def __init__(
         self.workspace = workspace
         self.include_package = include_package
 
-    def execute_step_graph(self, step_graph: StepGraph):
-        """
-        Execute a :class:`tango.step_graph.StepGraph`.
-        """
-        from tango import Step
-
+    def execute_step(self, step: "Step", is_uncacheable_leaf_step: bool = False) -> None:
         # Import included packages to find registered components.
         if self.include_package is not None:
             for package_name in self.include_package:
                 import_extra_module(package_name)
 
+        if step.cache_results:
+            step.ensure_result(self.workspace)
+        elif is_uncacheable_leaf_step:
+            step.result(self.workspace)
```

Comment:
Why do we have to check this? Whether it's leaf or not doesn't matter here, does it? When I call a method called `execute_step()`, I expect that either the step gets executed, or I get an exception. Under no circumstances would I expect the method to return after doing nothing.

Reply:
This is how it is currently run, and I have not changed the logic. Perhaps you can shed some light on [this](https://github.com/allenai/tango/commit/c46e0b2200012aad12e7ed232a3b076d979abeb9)? Perhaps it was needed then, and is now just a relic?

Reply:
Before your refactor, that code's responsibility was to execute a whole step graph. When you're doing that, it makes sense that you execute all cacheable steps, and also all uncacheable leaf steps. Interior uncacheable steps will be run automatically, but for uncacheable leaf steps we have to make sure they run manually, because they won't be run as part of some other step's dependencies.

Now, the name of the method promises that it will run a single step. In that context, it makes no sense to make that distinction.